### PR TITLE
add method Sort in args

### PR DESCRIPTION
**Pull request by xuecai**
sorted args string is useful in cache key。

### CODE REVIEW

**Comment by xuecai** [comment]
in your meas , it use like this ?
```
args := fasthttp.AcquireArgs()
// do something 

args.Sort() // just AppendBytesSort, it call sort.SliceStable and AppendBytes
cachekey := args.QueryString()  // call AppendBytes 
```

AppendBytesSort  do sort Args.args and rewrite Args.buf call AppendBytes
String and QueryString also call AppendBytes.
AppendBytes called twice is necessary ?

**Comment by erikdubbelboer** [comment]
Yes that is exactly what I mean. `Sort` should only do the sorting and have no argument or return value. That wat it's clear that you're changing the data by calling `Sort`. Right now it's not clear that you're changing the underlying data by calling `AppendBytesSort` for example.

**Comment by xuecai** [comment]
done , only Sort method add.
btw, Sort cost 3 allocs/op.

File: args.go [hunk 1]
```diff
@@ -115,6 +116,17 @@ func (a *Args) QueryString() []byte {
 	return a.buf
 }
 
+// Sort change args order which ASCII sorted by key, value.
```
Comment (erikdubbelboer) [inline args.go]: Can you please change this to:
```go
// Sort sorts Args lexicographically by key and then value.
```

Comment (xuecai) [reply args.go]: Args.args is not public , so maybe can't have the sort function.
I need time to think about it.

File: args.go [hunk 2]
```diff
@@ -115,6 +116,17 @@ func (a *Args) QueryString() []byte {
 	return a.buf
 }
 
+// Sort sorts Args lexicographically by key and then value.
```
Comment (erikdubbelboer) [inline args.go]: Good but of course now the comment should be
```go
// Sort sorts Args by key and then value using 'f' as comparison function.
//
// For example args.Sort(bytes.Compare)
```

**Comment by erikdubbelboer** [comment]
Looks good. Any comments @kirillDanshin ?

**Comment by xuecai** [comment]
@kirillDanshin   any comments?  my parterner want to use this feature . tks.

**Pull request merged**